controllers/humiocluster_controller.go

```diff
@@ -2349,3 +2426,184 @@ func getHumioNodePoolManagers(hc *humiov1alpha1.HumioCluster) HumioNodePoolList
 	}
 	return humioNodePools
 }
+
+// reconcileSinglePDB handles creation/update of a PDB for a single node pool
+func (r *HumioClusterReconciler) reconcileSinglePDB(ctx context.Context, hc *humiov1alpha1.HumioCluster, hnp *HumioNodePool) error {
+	pdbSpec := hnp.GetPodDisruptionBudget()
+	if pdbSpec == nil {
+		r.Log.Info("skipping PDB creation - node pool does not have PDB configured",
+			"nodePool", hnp.GetNodePoolName()) // Use NodePool Name here
+		return nil
+	}
+
+	pods, err := kubernetes.ListPods(ctx, r, hc.Namespace, kubernetes.MatchingLabelsForHumio(hc.Name))
+	if err != nil {
+		return fmt.Errorf("failed to list pods: %w", err)
+	}
+
+	if len(pods) == 0 {
+		r.Log.Info("no pods found, skipping PDB creation")
+		return nil
+	}
+
+	desiredPDB, err := r.constructPDB(hc, hnp, pdbSpec)
+	if err != nil {
+		r.Log.Error(err, "failed to construct PDB", "pdbName", hnp.GetPodDisruptionBudgetName(), "namespace", hnp.GetNamespace())
```

Comment:
One more place where you could skip adding the namespace yourself due to the logger already including this in all events.

Reply:
Removed "namespace" form the Log statement.